### DISCUSSION TITLE
License validity

### DISCUSSION THREAD

**Discussion by VaclavChalupa** [nager/Nager.Date#540] [General]
Hi,

When we choose the monthly commercial subscription, how long is the validity of the license key we will receive?

We are running some on-prem solutions, and monthly upgrades because of an expired license key is not feasible for us.
If the general validity is for one month, is there any possibility of receiving/paying for a license key with extended validity?

Thank you,

BR,

Vaclav

**Comment by tinohager** [comment]
Him the license key currently has no expiration date, if one is introduced then the license key will be valid for at least one year

**Comment by petrik-systemart** [comment]
What will happen when the old licence expires? Will the new license be available before the old one expires? Ideally at least a month earlier.

We have just become your sponsor and have written on the license to expire 2026-08-01

Thank you for the explanation

**Reply by tinohager** [reply]
The license key validity now depends on the selected sponsor tier.

You can find the specific validity period listed under the "Features" section of each tier. The license key is renewed on a monthly basis.

This allows you to decide when to update the key. It's recommended to renew it a few weeks before it expires to ensure uninterrupted access.

**Invoices**
Additionally, GitHub allows you to configure monthly invoices that are sent to you via email. This can be useful for accounting purposes.
![image](https://github.com/user-attachments/assets/d4afb448-0472-4b1c-aebb-85643819c277)

**Comment by XavierNV** [comment]
Hi Tino,
I got a message the licence key has expired (in my C# programm). But i'm still referenced as a Github sponsor - and haven't done any change concerning the sponsorship. Did the licence key was changed recently ? (I only checked today the "Receive email updates from nager". If so, how can i get the updated licence key ?
Thanks (and congrat for the project)
Xavier

**Reply by tinohager** [reply]
Hi @XavierNV

I have checked your case you are in the `Backer Easy` tier. Here there are no more license keys for the offline version until the commercial tiers. But since you are a loyal backer, you can send me an email and you will get a new key which will be valid for one year.

**Comment by XavierNV** [comment]
Hi Tino,
Thanks a lot for your answer. And your proposition... The commercial tiers is Ok for me, i will change the sponsorship for it...
Good luck with your project, Xavier
